### PR TITLE
feat: Add support for Client Credentials Grant authentication method

### PR DESCRIPTION
**Pull request by arjankowski**

### CODE REVIEW

File: Tests/BoxSDKSpecs.swift [hunk 1]
```diff
@@ -333,6 +333,177 @@ class BoxSDKSpecs: QuickSpec {
...
+                    sdk.getCCGClientForAccountService(enterpriseId: "987654321", tokenInfo: tokenInfo) { result in
+                        switch result {
+                        case let .success(ccgClient):
+                            client = ccgClient
+                            client?.session.getAccessToken { result in
+                                switch result {
+                                case let .success(token):
+                                    expect(token).to(equal(tokenInfo.accessToken))
+                                case let .failure(error):
+                                    fail("getCCGClientForUser should succeed, but instead got \(error)")
```
Comment (arjankowski) [inline Tests/BoxSDKSpecs.swift]: replace

File: Tests/BoxSDKSpecs.swift [hunk 2]
```diff
@@ -333,6 +333,177 @@ class BoxSDKSpecs: QuickSpec {
...
+                    sdk.getCCGClientForUser(userId: "654321", tokenStore: tokenStore) { result in
+                        switch result {
+                        case let .success(ouathClient):
+                            client = ouathClient
+                            client?.session.getAccessToken { result in
+                                switch result {
+                                case let .success(token):
+                                    expect(token).to(equal(tokenInfo.accessToken))
+                                case let .failure(error):
+                                    fail("getCCGClientForAccountService should succeed, but instead got \(error)")
```
Comment (arjankowski) [inline Tests/BoxSDKSpecs.swift]: replace

**Review by arjankowski — COMMENTED** [review]
Need to improve

**Review by antusus — APPROVED** [review]
👍

**Pull request merged**